alphafold3_pytorch/inputs.py

```diff
@@ -706,33 +744,32 @@ def alphafold3_input_to_molecule_input(
         len(all_protein_mols),
         len(all_rna_mols),
         len(all_dna_mols),
-        total_ligand_tokens
+        total_ligand_tokens,
     ]
 
     num_tokens = sum(molecule_type_token_lens) + num_metal_ions
 
-    assert num_tokens > 0, f'you have an empty alphafold3 input'
+    assert num_tokens > 0, "you have an empty alphafold3 input"
 
     arange = torch.arange(num_tokens)[:, None]
 
-    molecule_types_lens_cumsum = tensor([0, *molecule_type_token_lens]).cumsum(dim = -1)
+    molecule_types_lens_cumsum = tensor([0, *molecule_type_token_lens]).cumsum(dim=-1)
     left, right = molecule_types_lens_cumsum[:-1], molecule_types_lens_cumsum[1:]
 
+    # TODO: fix bug that may leave molecules with no assigned type
```

Comment:
i think it is just the metal ions (which are last)

Reply:
Oh ok